sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CheckpointFileManager.scala

```diff
@@ -83,6 +83,9 @@ trait CheckpointFileManager {
 
   /** Is the default file system this implementation is operating on the local file system. */
   def isLocal: Boolean
+
+  /** Returns a qualified path object for the file system's working directory. */
+  def makeQualified(path: Path): Path
```

Comment:
This trait has many APIs that take a `Path` parameter. Should it be a qualified path or a relative path? `CheckpointFileManager.create` takes the checkpoint location, which makes me think that all the APIs should take relative paths, implementation should qualify the path internally.

Reply:
Oh, that makes sense. Actually the place using `makeQualified` in `ResolveWriteToStream` is to call `mkdirs` on the qualified path. As now we also delegate to `CheckpointFileManager`, we don't need call `makeQualified` now.

Reply:
The qualified `resolvedCheckpointLocation` is put into `WriteToStream` and then used in `StreamExecution`. In the usage in `checkpointLocation`, it looks like it should be a qualified path, instead of a relative one.

I think we should keep it returned a qualified path in `ResolveWriteToStream`.

Reply:
I think it makes more sense to have
```
def getAbosolutePath(relativePath: Path)
```

It's weird to have an API to qualify the input path by file system's working directory, not the checkpoint location.

Reply:
or we add
```
def getQualifiedCheckpointPath(): Path
```

`CheckpointFileManager` already knows the checkpoint location.

Reply:
`getQualifiedCheckpointPath` sounds more reasonable here.